### PR TITLE
Try to receive interface in the receiver

### PR DESCRIPTION
**Pull request by tokahuke**
Backlog:

- [x] find a way to make things work without having to refactor everything.
- [x] implement `try_recv`, `try_rect_batch` and `try_recv_until`.
- [x] documentation for the new functions.
- [x] tests for the new functions.

### CODE REVIEW

**Comment by vedantroy** [comment]
I think `TryRecvError` needs to satisfy `Debug`, otherwise `unwrap` won't work.

**Comment by tokahuke** [comment]
(a) No, indeed, it needs to satisfy `Debug`. I will add that after work today.
(b) Yes and no.  Yes, the error types have to be public, but no, there is another way. I have reexported the errors to the crate's root, which has the same effect. 
(c) As you may have seen, I have found a bug in my implementation regarding long batches. The latest commit addresses that, but still needs tests to verify that it is now working correctly. Sorry that has sidetracked this PR a bit.
(d) Where did see the "queue" typo?

File: src/error.rs [hunk 1]
```diff
@@ -44,9 +44,44 @@ impl<T> TrySendError<T> {
...
                 base
             ),
         }
     }
 }
+
+/// An error that occurs when trying to receive from an empty queue.
+pub enum TryRecvError {
+    Io(io::Error),
+    QeueuEmpty, // { base: PathBuf }, problems with borrow checker. Leave it for future release...
```
Comment (vedantroy) [inline src/error.rs:57]: Queue typo.

Comment (tokahuke) [reply src/error.rs]: Ok. Solved.

**Pull request merged**